File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        
        int[] buckets = new int[101];

        for (int num : nums) {
          buckets[num]++;
        }

        for (int i = 1; i < buckets.length; i++) {
          buckets[i] += buckets[i - 1];
        }
        
        int[] result = new int[nums.length];
        for (int i = 0; i < result.length; i++) {
          if (nums[i] == 0)
            result[i] = 0;
          else
            result[i] = buckets[nums[i] - 1];
        }

        return result;

    }
}

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        size = len(nums)
        temp = [[num,ind] for ind,num in enumerate(nums)]
        temp.sort()
        prev = temp[0][0]
        temp[0][0] = 0
        nums[temp[0][1]] = 0
        for ind in range(1,size):
            if prev==temp[ind][0]:
                temp[ind][0] = temp[ind-1][0]
            else:
                prev = temp[ind][0]
                temp[ind][0] = ind
            nums[temp[ind][1]] = temp[ind][0]
            
        return nums
                
                